File: python/high-scores/high_scores.py
from typing import Sequence


class HighScores:
    def __init__(self, scores: Sequence[int]):
        self.scores = scores

    def latest(self) -> int:
        return self.scores[-1]

    def personal_best(self) -> int:
        return max(self.scores)

    def personal_top_three(self) -> list[int]:
        return sorted(self.scores, reverse=True)[:3]

File: exercises/practice/high-scores/.meta/src/reference/java/HighScores.java
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

class HighScores {

    private List<Integer> highScores;

    public HighScores(List<Integer> highScores) {
        this.highScores = new ArrayList<Integer>(highScores);
    }

    List<Integer> scores() {
        return new ArrayList<Integer>(highScores);
    }

    Integer latest() {
        return highScores.get(highScores.size() - 1);
    }

    Integer personalBest() {
        Optional<Integer> personalBest = highScores.stream().max(Comparator.naturalOrder());
        return personalBest.orElse(0);
    }

    List<Integer> personalTopThree() {
        return highScores.stream().sorted(Comparator.reverseOrder()).limit(3).collect(Collectors.toList());
    }

}
